src/components/colorscale/calc.js

```diff
@@ -20,7 +20,7 @@ module.exports = function calc(trace, vals, containerStr, cLetter) {
 
     if(containerStr) {
         container = Lib.nestedProperty(trace, containerStr).get();
-        inputContainer = Lib.nestedProperty(trace._input, containerStr).get();
+        inputContainer = Lib.nestedProperty(trace, containerStr).get();
```

Comment:
> My understanding is that we not set or overwrite what the user gaveth us;

Correct. But there are a few exceptions to the rule. Namely to make `restyle` work as desired. Too bad we never added any test cases. But, something like

```js
Plotly.plot(/* some graph with a colorscale and cmin/cmax auto filled */) 
Plotly.restyle(gd, 'cauto', false); 
```
should return the original graph, and at the moment to only way to do so is to mutate `gd.data` .